File: app/src/main/java/com/ulling/androidexsample/common/Define.java
package com.ulling.androidexsample.common;

import com.ulling.androidexsample.BuildConfig;
import com.ulling.lib.core.common.QcDefine;

/**
 * Class: Define
 * Version 1.0.0
 * Created by P139090 on 2018-12-28. 오후 3:35
 * <p>
 * Description:
 */
public class Define extends QcDefine {
    /**
     * 상용/ 테스트용
     */
    public static final boolean DEBUG_FLAG = BuildConfig.DEBUG;

    public static final int PAGE_SIZE = 20;

    public static final int HTTP_READ_TIMEOUT = 5000;
    public static final int HTTP_CONNECT_TIMEOUT = 4000;
    /**
     * TimeOut
     */
    public static final int INTRO_TIMEOUT = 1500;

    /**
     * time format
     */
    public static final String dateFormatFrom_sss = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
    public static final String dateFormatFrom_ss = "yyyy-MM-dd'T'HH:mm:ss'Z'";
    public static final String displayDateFormatFrom = "MM.dd aa hh:mm:ss";
    public static final String yyyyMMddFormatFrom = "yyyy MM.dd";


}

File: app/src/main/java/com/ulling/androidexsample/common/Define.kt
package com.ulling.androidexsample.common

import android.Manifest
import com.ulling.androidexsample.BuildConfig

class Define {

    companion object {
        /**
         * 상용/ 테스트용
         */
        val DEBUG_FLAG = BuildConfig.DEBUG

        const val PAGE_SIZE = 20

        const val HTTP_READ_TIMEOUT = 5000
        const val HTTP_CONNECT_TIMEOUT = 4000

        /**
         * TimeOut
         */
        const val INTRO_TIMEOUT = 1500

        /**
         * time format
         */
        const val dateFormatFrom_sss = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
        const val dateFormatFrom_ss = "yyyy-MM-dd'T'HH:mm:ss'Z'"
        const val displayDateFormatFrom = "MM.dd aa hh:mm:ss"
        const val yyyyMMddFormatFrom = "yyyy MM.dd"

    }
}

val  permissionList: Array<String> = arrayOf(
    Manifest.permission.WRITE_EXTERNAL_STORAGE,
//        Manifest.permission.ACCESS_NETWORK_STATE,
    Manifest.permission.CAMERA,
    Manifest.permission.RECORD_AUDIO
)

 val  permissionListAfterR: Array<String> = arrayOf(
    Manifest.permission.READ_EXTERNAL_STORAGE,
//        Manifest.permission.ACCESS_NETWORK_STATE,
    Manifest.permission.CAMERA,
    Manifest.permission.RECORD_AUDIO
)

 val  permissionListRW: Array<String> = arrayOf(
    Manifest.permission.WRITE_EXTERNAL_STORAGE,
    Manifest.permission.READ_EXTERNAL_STORAGE,
)